assets/js/main.js

```diff
@@ -0,0 +1,308 @@
+    cancelButtonColor: "#d33",
+    confirmButtonText: "Yes, delete it!",
+  }).then((result) => {
+    if (result.isConfirmed) {
+      tasksArray.splice(index, 1);
+      localStorage.setItem("tasksList", JSON.stringify(tasksArray));
+      displayData();
+      counter();
+
+      Swal.fire("Deleted!", "Your task has been deleted.", "success");
+    }
+  });
+}
+
+// To delete all tasks
+deleteAllButton.onclick = function () {
+  // sweet alert
+  Swal.fire({
+    title: "Are you sure?",
+    text: "You won't be able to revert this!",
+    icon: "warning",
+    showCancelButton: true,
+    confirmButtonColor: "#3085d6",
+    cancelButtonColor: "#d33",
+    confirmButtonText: "Yes, delete all !",
+  }).then((result) => {
+    if (result.isConfirmed) {
+      tasksArray.splice(0);
+      localStorage.setItem("tasksList", JSON.stringify(tasksArray));
+      displayData();
+      counter();
+      Swal.fire("Deleted!", "All task have been deleted.", "success");
+    }
+  });
+};
+
+// Search function
+function search(e) {
+  var result = "";
+  for (let i = 0; i < tasksArray.length; i++) {
+    if (tasksArray[i].name.toLowerCase().includes(e.value.toLowerCase())) {
+      result += `
+          <div class="taskDiv d-flex justify-content-between">
+      
+          <div class="taskAndAssignee">
+          <p class="taskNamePara pb-2">${tasksArray[i].name}</p>
+          <p class="assigneePara">${tasksArray[i].assignee}</p>
+          </div>
+      
+          <div class="icons">
+          <a onclick="deleteTask(${i})" class="deleteIcon d-block text-danger fs-5" href="#"><i class="fa fa-trash" aria-hidden="true"></i></a>
+          <a onclick="getTaskData(${i})" class="readIcon d-block text-warning fs-5" href="#"><i class="fa fa-id-card" aria-hidden="true"></i></a>
+          <a class="readIcon d-block text-success fs-5" href="#"><i class="fa fa-check-circle" aria-hidden="true"></i></a>
+          </div>
+      
+          </div>
+      
+          `;
+    }
+  }
+
+  debounceUpdate(result);
+}
+//Debounced search
+const debounceUpdate = debounce((result) => {
+  tasksData.innerHTML = result;
+});
+// debounce function
+function debounce(callback, delay = 1500) {
+  let timeout;
+  return (...args) => {
+    clearTimeout(timeout);
+    timeout = setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+// to display data in text filed
+function getTaskData(index) {
+  taskName.value = tasksArray[index].name;
+  assignee.value = tasksArray[index].assignee;
+  addTaskButton.innerHTML = "Update Task";
+  currentIndex = index;
+}
+// to update data in task div
+function updateTask() {
+  var taskObj = {
+    name: taskName.value,
+    assignee: assignee.value,
+  };
+  tasksArray[currentIndex].name = taskObj.name;
+  tasksArray[currentIndex].assignee = taskObj.assignee;
+
+  localStorage.setItem("tasksList", JSON.stringify(tasksArray));
+}
+// toggle button
+
+toggledBtn.addEventListener("click", function () {
```

Comment:
why did you use `addEventListener` and not `onClick`?

you should always avoid using listeners if possible because it affects performance. Listeners will keep running in the background listening to your changes, this is why we don't prefer using it if there is another way to perform the action you need to do.